endpoints/api/user.py

```diff
@@ -516,13 +516,16 @@ def post(self):
 
         # If recaptcha is enabled, then verify the user is a human.
         if features.RECAPTCHA:
-            recaptcha_response = user_data.get("recaptcha_response", "")
-            result = recaptcha2.verify(
-                app.config["RECAPTCHA_SECRET_KEY"], recaptcha_response, get_request_ip()
-            )
+            user = get_authenticated_user()
+            # check if the user is whitelisted to bypass recaptcha security check
+            if user is None or (user is not None and user.username not in features.RECAPTCHA_WHITELISTED_USERS):
```

Comment:
You don't need `user is not None` since it's a disjunction. Just `user is None or user.username not in features.RECAPTCHA_WHITELISTED_USERS`